launch/tier4_simulator_launch/launch/simulator.launch.xml

```diff
@@ -26,97 +26,100 @@
+        <arg name="use_base_link_z" value="$(var perception/use_base_link_z)"/>
+        <arg name="visible_range" value="$(var sensing/visible_range)"/>
+      </include>
+    </group>
 
-  <!-- perception module -->
-  <group>
-    <push-ros-namespace namespace="perception"/>
-    <!-- object recognition -->
-    <group if="$(var perception/enable_object_recognition)">
-      <push-ros-namespace namespace="object_recognition"/>
-      <!-- tracking module -->
-      <group>
-        <push-ros-namespace namespace="tracking"/>
-        <include file="$(find-pkg-share tier4_perception_launch)/launch/object_recognition/tracking/tracking.launch.xml">
-          <arg
-            name="object_recognition_tracking_radar_object_tracker_data_association_matrix_param_path"
-            value="$(var object_recognition_tracking_radar_object_tracker_data_association_matrix_param_path)"
-          />
-          <arg name="object_recognition_tracking_radar_object_tracker_tracking_setting_param_path" value="$(var object_recognition_tracking_radar_object_tracker_tracking_setting_param_path)"/>
-          <arg name="object_recognition_tracking_radar_object_tracker_node_param_path" value="$(var object_recognition_tracking_radar_object_tracker_node_param_path)"/>
-          <arg name="object_recognition_tracking_object_merger_data_association_matrix_param_path" value="$(var object_recognition_tracking_object_merger_data_association_matrix_param_path)"/>
-          <arg name="object_recognition_tracking_object_merger_node_param_path" value="$(var object_recognition_tracking_object_merger_node_param_path)"/>
-        </include>
-      </group>
-      <!-- prediction module -->
-      <group>
-        <push-ros-namespace namespace="prediction"/>
-        <include file="$(find-pkg-share tier4_perception_launch)/launch/object_recognition/prediction/prediction.launch.xml">
-          <arg name="use_vector_map" value="true"/>
-        </include>
-      </group>
+    <group unless="$(var scenario_simulation)">
+      <!-- Occupancy Grid -->
+      <push-ros-namespace namespace="occupancy_grid_map"/>
+      <include file="$(find-pkg-share tier4_perception_launch)/launch/occupancy_grid_map/probabilistic_occupancy_grid_map.launch.xml">
+        <arg name="input_obstacle_pointcloud" value="true"/>
+        <arg name="input_obstacle_and_raw_pointcloud" value="false"/>
+        <arg name="input/obstacle_pointcloud" value="/perception/obstacle_segmentation/pointcloud"/>
+        <arg name="output" value="/perception/occupancy_grid_map/map"/>
+        <arg name="occupancy_grid_map_method" value="laserscan_based_occupancy_grid_map"/>
+        <arg name="occupancy_grid_map_param_path" value="$(var laserscan_based_occupancy_grid_map_param_path)"/>
+        <arg name="occupancy_grid_map_updater" value="$(var occupancy_grid_map_updater)"/>
+        <arg name="occupancy_grid_map_updater_param_path" value="$(var occupancy_grid_map_updater_param_path)"/>
+      </include>
```

Comment:
OGM is disabled when launch_simulator_perception_modules is false? Is this the expected behavior?

Reply:
> We do not want to disable OGM actually.

So, how will you achieve the above? You will launch OGM by yourself outside the autoware.launch.xml?

Reply:
Yes, the variable names and conditions should represent what they really do.

OGM is a perception module so it should be disabled with the rest.

@brkay54 will launch it separately from the reaction analyzer node launch file.

If you have a better name for these variables or an alternative solution, we are open for it.

Reply:
I also agree to disable all the perception module with launch_simulator_perception_modules.

My question was that the PR's description said

> Therefore, we modify the launch file to disable all perception-related packages except occupancy_grid_map.

but actually, OGM is disabled as well.


Now I understand that you will launch it seprately.